src/realm/util/optional.hpp

```diff
@@ -119,6 +119,8 @@ class Optional : private _impl::OptionalStorage<T> {
     template <class... Args>
     void emplace(Args&&...);
     // FIXME: std::optional specifies an std::initializer_list overload for `emplace` as well.
+
+    void reset();
```

Comment:
This is the std-approved name: https://en.cppreference.com/w/cpp/utility/optional/reset